sdk/go/key_value/key-value.go

```diff
@@ -111,24 +104,29 @@ func fromCStrList(list *C.key_value_list_string_t) []string {
 
 	slice := unsafe.Slice(list.ptr, listLen)
 	for i := 0; i < listLen; i++ {
-		string := slice[i]
-		result = append(result, C.GoStringN(string.ptr, C.int(string.len)))
+		str := slice[i]
+		result = append(result, C.GoStringN(str.ptr, C.int(str.len)))
 	}
 
 	return result
 }
 
 func toErr(error *C.key_value_error_t) error {
 	switch error.tag {
-	case ErrorKindStoreTableFull: return errors.New("store table full")
-	case ErrorKindNoSuchStore: return errors.New("no such store")
-	case ErrorKindAccessDenied: return errors.New("access denied")
-	case ErrorKindInvalidStore: return errors.New("invalid store")
-	case ErrorKindNoSuchKey: return errors.New("no such key")
-	case ErrorKindIo: {
-		string := (*C.key_value_string_t)(unsafe.Pointer(&error.val))
-		return errors.New(fmt.Sprintf("io error: %s", C.GoStringN(string.ptr, C.int(string.len))))
-	}
-	default: return errors.New(fmt.Sprintf("unrecognized error: %v", error.tag))
+	case errorKindStoreTableFull:
+		return errors.New("store table full")
+	case errorKindNoSuchStore:
+		return errors.New("no such store")
+	case errorKindAccessDenied:
+		return errors.New("access denied")
+	case errorKindInvalidStore:
+		return errors.New("invalid store")
+	case errorKindNoSuchKey:
+		return errors.New("no such key")
```

Comment:
I realise this is not a change in behaviour but it does prompt me to ask how consumers of the Go API are meant to identify the "no such key" case (e.g. in order to default an initial value) - do they have to compare on the message?

Reply:
Great point.  They would need to check the error string.  I'm going to flip the approach and implement `Error` as an actual `error` and see how it feels.

Reply:
Creating the custom error is bloating this PR.  I'll create a new PR when it's completed